cmd/template-resolver/utils/templateResolver.go

```diff
@@ -0,0 +1,337 @@
+
+			objTemplates, _, err := unstructured.NestedSlice(objectDefinition, "spec", "object-templates")
+			if err != nil {
+				fmt.Fprintf(
+					os.Stderr,
+					"The ConfigurationPolicy at policy-templates index %d has an invalid object-templates array: %v\n",
+					i,
+					err,
+				)
+				os.Exit(1)
+			}
+
+			for _, objTemplate := range objTemplates {
+				jsonBytes, err := json.Marshal(objTemplate)
+				if err != nil {
+					fmt.Fprintf(
+						os.Stderr,
+						"The ConfigurationPolicy at policy-templates index %d has an invalid object-templates "+
+							"array: %v\n",
+						i,
+						err,
+					)
+					os.Exit(1)
+				}
+
+				rawDataList = append(rawDataList, jsonBytes)
+			}
+		}
+
+		objectTemplates := make([]interface{}, 0, len(rawDataList))
+
+		for _, rawData := range rawDataList {
+			if bytes.Contains(rawData, []byte("{{hub")) {
+				fmt.Fprintf(
+					os.Stderr,
+					"The ConfigurationPolicy at policy-templates index %d has an unresolved hub template. Use the "+
+						"-hub-kubeconfig argument.\n",
+					i,
+				)
+				os.Exit(1)
+			}
+
+			tmplResult, err := resolver.ResolveTemplate(rawData, nil, &resolveOptions)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to process the templates at policy-templates index %d: %v\n", i, err)
+				os.Exit(1)
+			}
+
+			var resolvedOT interface{}
+
+			err = json.Unmarshal(tmplResult.ResolvedJSON, &resolvedOT)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "Failed to process the templates at policy-templates index %d: %v\n", i, err)
+				os.Exit(1)
+			}
+
+			if oTRawFound {
+				switch v := resolvedOT.(type) {
+				case []interface{}:
+					objectTemplates = v
+				case nil:
+					objectTemplates = []interface{}{}
+				default:
+					fmt.Fprintf(
+						os.Stderr,
+						"object-templates-raw in policy-templates index %d was not an array after templates were "+
+							"resolved\n",
+						i,
+					)
+					os.Exit(1)
+				}
+
+				unstructured.RemoveNestedField(objectDefinition, "spec", "object-templates-raw")
+
+				break
+			}
+
+			objectTemplates = append(objectTemplates, resolvedOT)
+		}
+
+		err = unstructured.SetNestedSlice(objectDefinition, objectTemplates, "spec", "object-templates")
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Failed to process the templates at policy-templates index %d: %v\n", i, err)
+			os.Exit(1)
+		}
+	}
+
+	err = unstructured.SetNestedSlice(policy.Object, policyTemplates, "spec", "policy-templates")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "The resulting policy-templates were invalid: %v\n", err)
+		os.Exit(1)
+	}
+
+	resolvedPolicy, err := json.Marshal(policy.Object)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "The resulting Policy was invalid JSON: %v\n", err)
+		os.Exit(1)
+	}
+
+	resolvedYAML, err := templates.JSONToYAML(resolvedPolicy)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to convert the processed Policy back to YAML: %v\n", err)
+		os.Exit(1)
+	}
+
+	//nolint: forbidigo
+	fmt.Print(string(resolvedYAML))
```

Comment:
Since this is a public function, how about making this return the bytes/string of the resolved YAML and returns errors instead of exiting? Then the `main` function in `client.go` can handle printing the results, printing errors to stderr, and exiting when there is an error.

Reply:
Oh, good point!

Reply:
Good idea